### PR TITLE
Expose "token replace" as checkbox in the UI

### PR DESCRIPTION
**Pull request by nirgn975**
#83

### CODE REVIEW

File: src/Entity/MessageType.php [hunk 1]
```diff
@@ -169,6 +169,14 @@ public function setSettings(array $settings) {
     return $this;
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function setSetting($key, $setting) {
+    if (isset($this->settings[$key])) {
```
Comment (amitaibu) [inline src/Entity/MessageType.php]: `isset` can probably be removed

File: src/Form/MessageTypeForm.php [hunk 2]
```diff
@@ -130,6 +130,13 @@ public function form(array $form, FormStateInterface $form_state) {
       '#states' => $states,
     );
 
+    $form['settings']['token options']['token replace'] = array(
+      '#title' => t('Token replace'),
+      '#type' => 'checkbox',
+      '#description' => t('When this option is selected, token processing will happen.'),
+      '#default_value' => isset($settings['token options']['token replace']) ? $settings['token options']['token replace'] : TRUE,
```
Comment (amitaibu) [inline src/Form/MessageTypeForm.php]: enough just to do `'#default_value' =>  !empty($settings['token options']['token replace'])`

File: src/Form/MessageTypeForm.php [hunk 3]
```diff
@@ -130,6 +130,13 @@ public function form(array $form, FormStateInterface $form_state) {
       '#states' => $states,
     );
 
+    $form['settings']['token options']['token replace'] = array(
+      '#title' => t('Token replace'),
+      '#type' => 'checkbox',
+      '#description' => t('When this option is selected, token processing will happen.'),
```
Comment (amitaibu) [inline src/Form/MessageTypeForm.php:136]: those `t()` should probably be converted to `$this->t()`

File: src/Entity/MessageType.php [hunk 4]
```diff
@@ -169,6 +169,12 @@ public function setSettings(array $settings) {
     return $this;
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function setSetting($key, $setting) {
+    $this->settings[$key] = $setting;
```
Comment (RoySegall) [inline src/Entity/MessageType.php]: Missing 
`return $this`
at the end of the function.

File: src/MessageTypeInterface.php [hunk 5]
```diff
@@ -102,6 +102,11 @@ public function getText($langcode = Language::LANGCODE_NOT_SPECIFIED, $delta = N
   public function setSettings(array $settings);
 
   /**
+   * Set a single setting by key.
+   */
+  public function setSetting($key, $setting);
```
Comment (RoySegall) [inline src/MessageTypeInterface.php]: Missing `@return` param (after you'll add the `return $this`)

**Comment by nirgn975** [comment]
@RoySegall Ready for review.

**Comment by amitaibu** [comment]
Screenshot(s) please :)

File: src/MessageTypeInterface.php [hunk 6]
```diff
@@ -102,6 +102,13 @@ public function getText($langcode = Language::LANGCODE_NOT_SPECIFIED, $delta = N
   public function setSettings(array $settings);
 
   /**
+   * Set a single setting by key.
+   *
+   * @return array
+   */
+  public function setSetting($key, $setting);
```
Comment (amitaibu) [inline src/MessageTypeInterface.php]: missing `@param`s

**Comment by nirgn975** [comment]
@RoySegall @amitaibu Now it's really ready for review.

add_message_type___site-install

**Comment by amitaibu** [comment]
@nirgn975 needs a re-roll

**Comment by nirgn975** [comment]
@amitaibu Done.

File: src/Entity/MessageType.php [hunk 7]
```diff
@@ -172,6 +172,14 @@ public function setSettings(array $settings) {
   /**
    * {@inheritdoc}
    */
+  public function setSetting($key, $setting) {
```
Comment (amitaibu) [inline src/Entity/MessageType.php]: are we using this method?

**Comment by nirgn975** [comment]
@amitaibu Ready for review (I told u the tests will be green ;)

File: src/Form/MessageTypeForm.php [hunk 8]
```diff
@@ -130,6 +130,13 @@ public function form(array $form, FormStateInterface $form_state) {
       '#states' => $states,
     ];
 
+    $form['settings']['token options']['token replace'] = array(
+      '#type' => 'checkbox',
+      '#title' => t('Token replace'),
+      '#description' => t('When this option is selected, token processing will happen.'),
+      '#default_value' => isset($settings['token options']['token replace']) ? !empty($settings['token options']['token replace']) : TRUE,
```
Comment (amitaibu) [inline src/Form/MessageTypeForm.php]: the `!empty()` seems wrong, it should probably be:

`'#default_value' =>  !isset($settings['token options']['token replace']) || !empty($settings['token options']['token replace'])`

**Comment by nirgn975** [comment]
@amitaibu Ready for code review.

**Comment by amitaibu** [comment]
Thanks.

File: message.api.php [hunk 9]
```diff
@@ -1,137 +0,0 @@
...
-  return $defaults;
-}
-
-/**
- * Alter default message type category configurations.
- *
- * @param array $defaults
- *   An array of default message type categories, keyed by machine names.
- *
- * @see hook_default_message_category()
```
Comment (Damacgod1100) [inline message.api.php:129]: Leave a line note

**Pull request merged**